### PR TITLE
Add email template for TAM event type

### PR DESCRIPTION
**Pull request by astrozzc**
Adding email templates for new event type: rh-new-tam-request-created.

https://issues.redhat.com/browse/RHCLOUD-25247

### CODE REVIEW

File: engine/src/test/java/com/redhat/cloud/notifications/templates/TestRbacTemplate.java [hunk 1]
```diff
@@ -115,6 +116,9 @@ private void testTitle(String eventType, String result) {
             case REQUEST_ACCESS:
                 assertEquals("Instant notification - Request access - User Access - Console", result);
                 break;
+            case TAM_ACCESS_REQUESTED:
```
Comment (g-duval) [inline engine/src/test/java/com/redhat/cloud/notifications/templates/TestRbacTemplate.java:119]: Could you please also consider to add a test case on `testBody` method ?

File: engine/src/main/resources/templates/Rbac/tamAccessRequestEmailTitleV2.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+Instant notification - A technical account manager requested to access your account - User Access - Console
```
[the remark this answers is not in the document]
Comment (astrozzc) [reply engine/src/main/resources/templates/Rbac/tamAccessRequestEmailTitleV2.txt:1]: Thanks, updated

File: engine/src/main/resources/templates/drawer/Rbac/tamAccessRequest.html [hunk 3]
```diff
@@ -0,0 +1 @@
+A Techenical Account Manager <b>{data.events[0].payload.username}</b> requested to access your account.
```
[the remark this answers is not in the document]
Comment (astrozzc) [reply engine/src/main/resources/templates/drawer/Rbac/tamAccessRequest.html:1]: Good catch!

File: engine/src/main/java/com/redhat/cloud/notifications/templates/DrawerTemplateMigrationService.java [hunk 4]
```diff
@@ -229,6 +229,10 @@ public List<String> migrate() {
                 warnings, "console", "rbac", List.of("request-access"),
                 "Rbac/requestAccess", "html", "Rbac platform group to custom drawer body"
         );
+        createDrawerIntegrationTemplate(
+                warnings, "console", "rbac", List.of("rh-new-tam-request-created"),
+                "Rbac/tamAccessRequest", "html", "Rbac tam access request drawer body",
```
[the remark this answers is not in the document]
Comment (astrozzc) [reply engine/src/main/java/com/redhat/cloud/notifications/templates/DrawerTemplateMigrationService.java:234]: Updated

File: engine/src/test/java/com/redhat/cloud/notifications/templates/TestRbacTemplate.java (hunk 1 above)
Comment (astrozzc) [reply engine/src/test/java/com/redhat/cloud/notifications/templates/TestRbacTemplate.java:119]: Sure, just added

File: engine/src/test/java/com/redhat/cloud/notifications/templates/TestRbacTemplate.java [hunk 5]
```diff
@@ -115,6 +116,9 @@ private void testTitle(String eventType, String result) {
             case REQUEST_ACCESS:
                 assertEquals("Instant notification - Request access - User Access - Console", result);
                 break;
+            case RH_TAM_ACCESS_REQUESTED:
+                assertEquals("Instant notification - TAM access requested - User Access - Console", result);
```
Comment (g-duval) [inline engine/src/test/java/com/redhat/cloud/notifications/templates/TestRbacTemplate.java:120]:
```suggestion
                assertEquals("Instant notification - New TAM access request created - User Access - Console", result);
```
It have to be the exact same content as `engine/src/main/resources/templates/Rbac/tamAccessRequestEmailTitleV2.txt`

**Review by g-duval — CHANGES_REQUESTED** [review]

File: engine/src/test/java/com/redhat/cloud/notifications/templates/TestRbacTemplate.java (hunk 5 above)
Comment (astrozzc) [reply engine/src/test/java/com/redhat/cloud/notifications/templates/TestRbacTemplate.java]: Oh, thanks for catching that! I looked through the log but did not find where it was failing. Updated

**Review by g-duval — APPROVED** [review]

**Comment by g-duval** [comment]
error on ci.ext.devshift.net PR build is unrelated to this PR

**Pull request merged**